libs/client/cornflow_client/constants.py

```diff
@@ -90,6 +90,74 @@
     pl.LpStatusUndefined: STATUS_UNDEFINED,
 }
 
+PARAMETER_SOLVER_TRANSLATING_MAPPING = {
+    ('bar_iter_limit', 'pyomo', 'cbc') : 'barr' ,
+    ('best_obj_stop', 'pyomo', 'cbc') : 'primalT' ,
+    ('cutoff', 'pyomo', 'cbc') : 'cuts' ,
+    ('iteration_limit', 'pyomo', 'cbc') : 'maxIt' ,
+    ('solution_limit', 'pyomo', 'cbc') : 'maxSaved' ,
+    ('time_limit', 'pyomo', 'cbc') : 'sec' ,
+    ('mip_gap', 'pyomo', 'cbc') : 'allow' ,
+    ('optimality_tol', 'pyomo', 'cbc') : 'dualT' ,
+    ('pump_passes', 'pyomo', 'cbc') : 'pumpC' ,
+    ('heuristics', 'pyomo', 'cbc') : 'heur' ,
+
+    ('is_mip', 'pulp', 'cbc') : 'mip' ,
+    ('optimality_tol', 'pulp', 'cbc') : 'gapAbs' ,
+    ('cutoff', 'pulp', 'cbc') : 'cuts' ,
+    ('time_limit', 'pulp', 'cbc') : 'timeLimit' ,
+    ('threads', 'pulp', 'cbc') : 'threads' ,
+    ('presolve', 'pulp', 'cbc') : 'presolve' ,
+
+    ('bar_iter_limit', 'pyomo', 'gurobi') : 'BarIterLimit' ,
+    ('best_obj_stop', 'pyomo', 'gurobi') : 'BestObjStop' ,
+    ('cutoff', 'pyomo', 'gurobi') : 'Cutoff' ,
+    ('iteration_limit', 'pyomo', 'gurobi') : 'IterationLimit' ,
+    ('mem_limit', 'pyomo', 'gurobi') : 'MemLimit' ,
+    ('solution_limit', 'pyomo', 'gurobi') : 'SolutionLimit' ,
+    ('time_limit', 'pyomo', 'gurobi') : 'TimeLimit' ,
+    ('mip_gap', 'pyomo', 'gurobi') : 'MIPGap' ,
+    ('optimality_tol', 'pyomo', 'gurobi') : 'OptimalityTol' ,
+    ('branch_dir', 'pyomo', 'gurobi') : 'BranchDir' ,
+    ('pump_passes', 'pyomo', 'gurobi') : 'PumpPasses' ,
+    ('heuristics', 'pyomo', 'gurobi') : 'Heuristics' ,
+    ('presolve', 'pyomo', 'gurobi') : 'Presolve' ,
+    ('threads', 'pyomo', 'gurobi') : 'threads' ,
+
+    ('is_mip', 'pulp', 'gurobi') : 'mip' ,
+    ('time_limit', 'pulp', 'gurobi') : 'timeLimit' ,
+
+    ('bar_tol', 'pyomo', 'scip') : 'numerics/barrierconvtol' ,
+    ('cutoff_breaker', 'pyomo', 'scip') : 'heuristics/shiftandpropagate/cutoffbreaker' ,
+    ('lp_iteration_limit', 'pyomo', 'scip') : 'lp/iterlim' ,
+    ('mem_limit', 'pyomo', 'scip') : 'limits/memory' ,
+    ('solution_limit', 'pyomo', 'scip') : 'limits/maxsol' ,
+    ('time_limit', 'pyomo', 'scip') : 'limits/time' ,
+    ('primal-dual_gap', 'pyomo', 'scip') : 'limits/gap' ,
+    ('lp_optimality_tol', 'pyomo', 'scip') : 'numerics/lpfeastol' ,
+    ('pump_passes', 'pyomo', 'scip') : 'heuristics/feaspump/maxdepth' ,
+    ('presolve', 'pyomo', 'scip') : 'presolve' ,
+    ('threads', 'pyomo', 'scip') : 'threads' ,
+
+    ('is_mip', 'pulp', 'scip') : 'mip' ,
+    ('optimality_tol', 'pulp', 'scip') : 'gapAbs' ,
+    ('time_limit', 'pulp', 'scip') : 'timeLimit' ,
+    ('threads', 'pulp', 'scip') : 'threads' ,
+    ('max_nodes', 'pulp', 'scip') : 'maxNodes' ,
```

Comment:
How do you set up the `optimality_tol` in pyomo?

Reply:
But the name should be the same that the one used in pulp, no? Why it is `optimality_tol` in one and `lp_optimality_tol` in the other?

Reply:
This is outdated. The most updated version is where you commented about optimality_tol and primal-dual_gap.